OS X 10.8 Modifications are in

ma-ku opened the pull request:
Here is my contribution to bluecove that allows using the library with the latest version of OS X (10.8 aka Mountain Lion). I could not test this version with any older version of OS X so if this may become an issue it may be easier to have the old and the new version side by side.

ma-ku committed:
- Added .gitignore to allow clean commits

ma-ku committed:
Modified bluecove to work with OS X 10.8

The main part of the code was adapted to use the new bluetooth APIs introduced by Apple and which were enforced with the release of OS X 10.8 (aka Mountain Lion). The modified code facilitates GCD and semaphores to overcome the (also) deprecated concurrency APIs. However, still some function calls have no counterpart in the new API and were disabled for the moment.

ma-ku committed:
Added .gitignore to exclude the maven artifacts

ma-ku committed:
Some more exclusions

ma-ku committed:
More exclusions to get rid of Xcode artifacts

ma-ku committed:
Fixed problems with string conversion

ma-ku committed:
Modified project POM to reflect contribution

ma-ku committed:
Updated project to reflect latest build environment (OS X)

ma-ku committed:
Streamlined ignores (finally)

ieee8023 commented:
Hey. Thanks for making it work on 10.8. However, I cannot compile the ma-ku version on 10.6 as you predicted.  Maybe it's best to treat it as a whole new stack so they are different compile targets?  Here is the error I get:

"error: There is no SDK with the name or path 'macosx10.7'"

Let me know and I can spend some time doing something.

-J

ma-ku commented:
Usually you can switch to another SDK in the project configuration. Just search for the term 'Base SDK' in the build settings (select the blue icon in the left tree view and switch to the build settings tab on the right side). There you can select the correct SDK.

Cheers

Mathias

hcarver commented:
Thanks for this and sorry it's taken me so long to get around to it. Please could you build bluecove with your new code and put the result online somewhere I can download it, so that I can test whether it's backwards-compatible with older releases of OS X?

Commit:
Fix crash due to incorrect array initialization.

Commit:
Fixed crash due to incorrect allocation of NSDate. Fixed some logical errors from the OS X 10.8 update.

Commit:
Fixed a leak.

Commit:
Fixed regression with OSX 10.8 update due to incorrect use of dispatch_semaphore_wait return value.

Commit:
Fixed regression in OSX 10.8 update (deadlock when dispatching BTOperation on main thread). Also removed unnecessary semaphore.

ma-ku committed:
Fixed issues with latest SDK
Corrected invalid parameter handling in getLocalDeviceVersion()
Corrected exception raised by invalid reference counting

littlefreaky committed:
set failonerror to false in ant delete task (clean-remove-created-files-main) to prevent build failure if the src/main/resources directory does not exist

littlefreaky committed:
- Implemented the function Java_com_intel_bluetooth_BluetoothStackOSX_retrieveDevicesImpl with support to retrieve PREKNOWN bluetooth devices
- Added dummy implementation for Java_com_intel_bluetooth_BluetoothStackOSX_isRemoteDeviceTrustedImpl and Java_com_intel_bluetooth_BluetoothStackOSX_isRemoteDeviceAuthenticatedImpl to prevent UnstatisfiedLinkErrors

littlefreaky committed:
Revert "- Implemented the function Java_com_intel_bluetooth_BluetoothStackOSX_retrieveDevicesImpl with support to retrieve PREKNOWN bluetooth devices - Added dummy implementation for Java_com_intel_bluetooth_BluetoothStackOSX_isRemoteDeviceTrustedImpl and Java_com_intel_bluetooth_BluetoothStackOSX_isRemoteDeviceAuthenticatedImpl to prevent UnstatisfiedLinkErrors "

This reverts commit 73880b49f8cd5d33971104fe168fcf0c0eaecb6f.

littlefreaky committed:
Revert "set failonerror to false in ant delete task (clean-remove-created-files-main) to prevent build failure if the src/main/resources directory does not exist"

This reverts commit 979c47afc4f5936b9992d78fade85f6f7724ea42.

littlefreaky committed:
- Implemented the function Java_com_intel_bluetooth_BluetoothStackOSX_retrieveDevicesImpl with support to retrieve PREKNOWN bluetooth devices
- Added dummy implementation for Java_com_intel_bluetooth_BluetoothStackOSX_isRemoteDeviceTrustedImpl and Java_com_intel_bluetooth_BluetoothStackOSX_isRemoteDeviceAuthenticatedImpl to prevent UnstatisfiedLinkErrors

ma-ku committed:
Merge pull request #3 from revite/master

Fixed regression in OSX 10.8 update (deadlock when dispatching BTOperati...

ma-ku committed:
Merge branch 'master' of https://github.com/ma-ku/bluecove

ma-ku committed:
- Modified base SDK
